bigchaindb/backend/mongodb/changefeed.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import time
+
+import pymongo
+from pymongo.errors import ConnectionFailure, OperationFailure
+
+from bigchaindb import backend
+from bigchaindb.backend.changefeed import ChangeFeed
+from bigchaindb.backend.utils import module_dispatch_registrar
+from bigchaindb.backend.mongodb.connection import MongoDBConnection
+
+
+logger = logging.getLogger(__name__)
+register_changefeed = module_dispatch_registrar(backend.changefeed)
+
+
+class MongoDBChangeFeed(ChangeFeed):
+    """This class implements a MongoDB changefeed.
+
+    We emulate the behaviour of the RethinkDB changefeed by using a tailable
+    cursor that listens for events on the oplog.
+    """
+
+    def run_forever(self):
+        for element in self.prefeed:
+            self.outqueue.put(element)
+
+        while True:
+            try:
+                self.run_changefeed()
+                break
+            except (ConnectionFailure, OperationFailure) as exc:
+                logger.exception(exc)
+                time.sleep(1)
+
+    def run_changefeed(self):
+        # this is kinda a hack to make sure that the connection object is
+        # setup
+        self.connection._connect()
+        # namespace to allow us to only listen to changes in a single
+        # collection
+        dbname = self.connection.dbname
+        table = self.table
+        namespace = '{}.{}'.format(dbname, table)
+        # last timestamp in the oplog. We only care for operations happening
+        # in the future.
+        last_ts = self.connection.conn.local.oplog.rs.find()\
+                      .sort('$natural', pymongo.DESCENDING).limit(1)\
+                      .next()['ts']
+        # tailable cursor. A tailable cursor will remain open even after the
+        # last result was returned. ``TAILABLE_AWAIT`` will block for some
+        # timeout after the last result was returned. If no result is received
+        # in the meantime it will raise a StopIteration excetiption.
+        cursor = self.connection.conn.local.oplog.rs.find(
+            {'ns': namespace, 'ts': {'$gt': last_ts}},
+            cursor_type=pymongo.CursorType.TAILABLE_AWAIT
+        )
+
+        while cursor.alive:
+            try:
+                record = cursor.next()
+            except StopIteration:
+                continue
+            else:
```

Comment:
since we have a `continue` in the `except`, do we need the `else` here? If now, we can de-indent the next lines by one block.